dali/kernels/imgproc/warp/map_coords.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_WARP_MAP_COORDS_H_
+#define DALI_KERNELS_IMGPROC_WARP_MAP_COORDS_H_
+
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/imgproc/sampler.h"
+#include "dali/kernels/imgproc/warp/warp_setup.cuh"
+#include "dali/kernels/imgproc/warp/mapping_traits.h"
+
+namespace dali {
+namespace kernels {
+namespace warp {
+
+/** @brief Converts destination _integer_ coordinates to _floating point_ source coordinates
+ *
+ * This function is used by warp implementations to calculate source coordinates using
+ * using non-integer mapping. The coordinates are converted to pixel-centered.
+ */
+template <typename Mapping, std::size_t dim>
```

Comment:
use `int dim`

Reply:
I must use whatever vec is using.

Reply:
I'm considering changing `vec` and `mat` to use `int` instead - they are designed to be _small_ vectors/matrices, so putting a large size_t makes little sense.